sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -265,7 +265,15 @@ class SparkSession private(
    * implementation is Hive, this will initialize the metastore, which may take some time.
    */
   private[sql] def cloneSession(): SparkSession = {
+    val existingSessionConf = sessionState.conf.getAllConfs
```

Comment:
Different question, why are these confs not copied over the in the `BaseSessionStateBuilder`?

Reply:
I just checked, I think we should fix this in `BaseSessionStateBuilder.conf`, we should only copy the SparkConf settings if the `parentState` is None.

Reply:
+1, good point.

Reply:
@hvanhovell  @gaborgsomogyi  Thanks for reviewing . and Yes,that should be the correct fix.